geth/node/manager.go

```diff
@@ -502,8 +502,36 @@ func (m *NodeManager) AccountKeyStore() (*keystore.KeyStore, error) {
 	return keyStore, nil
 }
 
-// RPCClient exposes reference to RPC client connected to the running node.
-func (m *NodeManager) RPCClient() (*rpc.Client, error) {
+// RPCLocalClient exposes reference to RPC client connected to the running node.
+func (m *NodeManager) RPCLocalClient() (*rpc.Client, error) {
+	if m == nil {
+		return nil, ErrInvalidNodeManager
+	}
+
+	m.RLock()
+	defer m.RUnlock()
+
+	// make sure that node is fully started
+	if m.node == nil || m.nodeStarted == nil {
+		return nil, ErrNoRunningNode
+	}
+
+	<-m.nodeStarted
```

Comment:
Are you sure we need all these checks?

Reply:
In my opinion, the `m.nodeStarted` is abit overkill but the checks are necessary even here.

Reply:
I have an inner desire to remove all that as it's unrelated to what this function does but I'm not yet sure about the best way to accomplish it. Let's leave it for now.

Reply:
Alright. I cant remember who, but someone did re-write the checks into a single method, am sure it is in one of the PRs.

Reply:
I took over that PR but it might not get merged as I tested the issue with the current develop branch and everything worked as expected, meaning it looks like no changes are needed.

Reply:
Ok. We can discuss this a little then refactor in another issue, as @tiabc has thoughts on this.